src/Swift.Runtime/src/Library/UnsafePointer.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Swift.Runtime
+{
+    // <summary>
+    // Represents Swift UnsafePointer in C#.
+    // </summary>
+    public unsafe readonly struct UnsafePointer<T>
```

Comment:
I think it would make sense to have implicit conversions to/from T* for these types.

Reply:
(And to/from `void*` for the non-generic variant.)